client/http.go

```diff
@@ -0,0 +1,88 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/drand/drand/beacon"
+	"github.com/drand/drand/key"
+)
+
+// NewHTTPClient creates a new client pointing to an HTTP endpoint
+func NewHTTPClient(url string, groupHash []byte, client *http.Client) (Client, error) {
```

Comment:
I recommend defining an interface with the `http.Client` methods required by this type and accept that here instead, rather than requiring the concrete type. The concrete type is very difficult to replace in testing. It will be difficult to test the non-trival logic in this type without exposing yourself to network flakiness and use of a full-on HTTP server on the other end, otherwise. 

 Also, integration tests in consumers may wish to use as much "real" drand code client as possible, but not involve any actual network requests, and so a fake http client with directly configurable group info and randomness can be substituted here. (This is slightly less important because consumer tests can use a fake client.Client impl instead and avoid this object entirely).